# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/VideoConduit.h
# Commit: b2332c1d550e
# Full Hash: b2332c1d550ed2d02408150e9979685981d0817f
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 5
# Description:
#   Bug 1654112 - Don't push RtpSource info to main thread on every received frame. r=ng,bwc
#   
#   If js doesn't intend to look at get*Sources, then pushing RtpSource info to main
#   thread constantly is cycles spent in vain. Considering that upstream already
#   updates this info off-main on every received frame it is also to a large extent
# ==============================================================================

diff -r c186df8a088e -r b2332c1d550e dom/media/webrtc/libwebrtcglue/VideoConduit.h
--- a/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Fri Aug 20 13:52:08 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Fri Aug 20 14:03:36 2021 +0200
@@ -98,8 +98,6 @@
   Maybe<DOMHighResTimeStamp> LastRtcpReceived() const override;
   DOMHighResTimeStamp GetNow() const override;
 
-  void OnFrameDelivered() override;
-
   MediaConduitErrorCode StopTransmittingLocked();
   MediaConduitErrorCode StartTransmittingLocked();
   MediaConduitErrorCode StopReceivingLocked();
@@ -228,6 +226,8 @@
   void OnRtcpBye() override;
   void OnRtcpTimeout() override;
 
+  std::vector<webrtc::RtpSource> GetUpstreamRtpSources() const override;
+
  private:
   // Don't allow copying/assigning.
   WebrtcVideoConduit(const WebrtcVideoConduit&) = delete;